fix(ErpNext): use correct endpoint to fetch DocType fields with proper access control

Mehul945 opened the pull request:
### **Summary**

This PR updates the logic to use the correct Frappe API endpoint for retrieving document fields metadata. Previously, the endpoint being used (`/api/resource/DocType`) only allowed administrator-level access, which caused permission issues for non-admin users.

With this update:

* We now use an endpoint that includes appropriate access control and returns full DocType metadata, including parent and child tables.
* This ensures that document field access is permission-aware and works for all authorized users.

To test:

* Run the workflow as a non-admin user and confirm fields load successfully.
* Confirm child table fields are included.

---

### **Related Linear tickets, Github issues, and Community forum posts**

<!-- Update this section if applicable -->

*No related tickets.*

---

### **Review / Merge checklist**

* [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md))
* [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
* [x] Tests included.
* [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

Mehul945 committed:
fix(ERPNext): Update API call for fetching document fields. Ensure proper access control and permissions are applied

Joffcom commented:
Hey @Mehul945,

Thanks for the PR, We have created "GHC-3074" as the internal reference to get this reviewed.

One of us will be in touch if there are any changes needed, in most cases this is normally within a couple of weeks but it depends on the current workload of the team.

Mehul945 commented:
Thanks for the update!
Noted the internal reference: GHC-3074.

Let me know if anything needs to be changed or clarified — happy to help. Looking forward to the review outcome.

Joffcom commented:
Hey @Mehul945,

The only 2 outstanding things looks to be signing the CLA so we can accept the PR and adding tests.

Mehul945 commented:
Hi @Joffcom ,

I’ve signed the CLA — please let me know if anything else is needed.

Thanks!

Mehul945 commented:
Hi @michael-radency, just following up on this PR — it’s been open for a few days, and I’d really appreciate a review when you get a chance. Let me know if there’s anything I can clarify or improve. Thanks!

Mehul945 committed:
Merge branch 'master' into master

Mehul945 commented on packages/nodes-base/nodes/ERPNext/ERPNext.node.ts:
```diff
@@ -92,18 +92,19 @@ export class ERPNext implements INodeType {
+					`/api/method/frappe.desk.form.load.getdoctype?doctype=${docType}`,
 					{},
 				);
 
```
@michael-radency  can you please response, I'm waiting to for your response.

Mehul945 committed:
Merge branch 'master' into master

michael-radency committed:
Merge branch 'master' into master

michael-radency reviewed (changes requested):
Hello @Mehul945 , sorry for the late reply. A few things need to be addressed first:

* As the change could be breaking, please gate it behind the new node version.
* Please add tests to ensure old behaviour is preserved and the new version is covered.
* Please address the automated review comment.

Mehul945 committed:
Merge branch 'n8n-io:master' into master

Mehul945 commented:
Addressed all requested changes:
- Added ERPNext node version 1.1.
- Preserved the existing version 1 behavior.
- Added a Frappe source reference for the undocumented endpoint.
- Safely handled malformed API responses.
- Added tests for both versions, child fields, and malformed responses.
Validation: 4 tests passed, targeted ESLint passed, and Biome pre-commit checks passed.
